Experimental support for AIRWELL protocol.

crankyoldgit opened the pull request:
* Add `matchManchester()` to IRrecv class.
* Add `sendManchester()` to IRsend class.
* Add `sendAirwell()` & `decodeAirwell`.
* Update support functions.
* Add unit tests.

Fixes #1069

crankyoldgit committed:
Experimental support for AIRWELL protocol.
* Add `matchManchester()` to IRrecv class.
* Rough versions of `sendAirwell()` & `decodeAirwell`.
  - These may/probably will change. i.e. Don't count on the values being
the same.
* Update support functions.
* Add unit tests.

For #1069

crankyoldgit committed:
Update SupportedProtocols.md

crankyoldgit committed:
Add a generic `sendManchester()` method.
Generic methods to send machester based codes.
This should enable some code reduction by using common code amoungst all
the protocols that use it.

Unit tests added to make sure it works.
Hand checked against: https://en.wikipedia.org/wiki/Manchester_code

jamsinclair commented on test/ir_Airwell_test.cpp:
```diff
@@ -0,0 +1,133 @@
+TEST(TestDecodeAirwell, RealExample) {
+  IRsendTest irsend(kGpioUnused);
+  IRrecv irrecv(kGpioUnused);
+  // ON / OFF / 25 degres heat
```
spelling: degrees

crankyoldgit replied:
Yep. That will teach me to cut & paste from a user's comment.

Fixed.

jamsinclair commented on tools/Makefile:
```diff
@@ -51,7 +51,8 @@ PROTOCOLS = ir_NEC.o ir_Sony.o ir_Samsung.o ir_JVC.o ir_RCMM.o ir_RC5_RC6.o \
             ir_MitsubishiHeavy.o ir_Goodweather.o ir_Inax.o ir_Argo.o \
-						ir_Trotec.o ir_Neoclima.o ir_Amcor.o ir_Epson.o ir_Symphony.o
+						ir_Trotec.o ir_Neoclima.o ir_Amcor.o ir_Epson.o ir_Symphony.o \
+            ir_Airwell.o
```
Unrelated to PR, but looks like this file has mixed spaces and tabs?

crankyoldgit replied:
Fixed.

crankyoldgit committed:
Changes based on code review feedback from @jamsinclair

crankyoldgit committed:
use `sendManchester()` in AIRWELL

crankyoldgit committed:
Progress check-in
Fairly sure sendManchester() & matchManchester() are working.

Still need to fixup Airwell protocol.

crankyoldgit committed:
Complete re-write of `matchManchester()`
* Couldn't solve the bugs and the code was too messy. Did a complete
re-write from scratch.
* Remove debug messages.
* Add more testing.
* Simplify `sendAirwell()`
* Adjust timings in `decodeAirwell()`
* Set default repeat for AIRWELL.

For #1069

crankyoldgit commented:
@jamsinclair Hey Jamie, can I get you to please take another look? I found some stubborn bugs in some further testing I added. Thus I had to basically re-write all of `matchManchester()` again.

Hence, the request for a re-review. Sorry!

crankyoldgit committed:
matchManchester: Adjust footer mark matching.

* match the UNKNOWN msg in 
https://github.com/crankyoldgit/IRremoteESP8266/issues/1069#issuecomment-607659677

For #1069

crankyoldgit committed:
Fix linter error

crankyoldgit commented:
FYI, sending has been confirmed to be working by @skyderman in: https://github.com/crankyoldgit/IRremoteESP8266/issues/1069#issuecomment-607676531
Decoding appears to be (mostly) working too.

crankyoldgit committed:
Airwell: Update to BETA status.
Reports are it appears to be working as expected.
For #1069

jamsinclair reviewed (approved):
Looks good, great work debugging this all!

crankyoldgit commented:
@jamsinclair Thanks for the prompt re-review. Appreciated!
